scripts/gulp/frontend-shared.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const cp = require('child_process');
+const fs = require('fs');
+const gulp = require('gulp');
+const babel = require('gulp-babel');
+const sourcemaps = require('gulp-sourcemaps');
+
+const buildFrontendSharedJs = () => {
+  // There does not appear to be a simple way of forcing gulp-babel to use a config
+  // file. Load it up as JSON and pass it in manually.
+  const babelConfig = JSON.parse(
+    fs.readFileSync('./frontend-shared/babel.config.json')
+  );
+
+  return (
+    gulp
+      .src('frontend-shared/src/*')
+      // Transpile the js source files and write the output in the frontend-shared/lib dir.
+      // Additionally, add the sourcemaps into the same dir.
+      .pipe(sourcemaps.init())
+      .pipe(babel(babelConfig))
+      .pipe(sourcemaps.write('.'))
+      .pipe(gulp.dest('frontend-shared/lib'))
+  );
+};
+
+const linkFrontendShared = done => {
+  // Setup a symlink from the client to the frontend-shared package.
+  cp.spawn('yarn', ['link'], {
```

Comment:
You can use [util.promisify](https://nodejs.org/dist/latest-v8.x/docs/api/util.html#util_util_promisify_original) to cut out the callback chain here.

Reply:
I tried to tinker around with that, but I'm failing to see how to improve readability. The best I could do is create a custom promise that also just watches for the 'end' events, but that increases complexity. What do you suggest?

Reply:
Ah right. I was thinking of typical Node async functions which run and invokes a callback with `(err, result)` arguments when it completes. `spawn` isn't like that.